partition_processor.go

```diff
@@ -188,33 +190,24 @@ func (pp *PartitionProcessor) Recovered() bool {
 	return pp.state.IsState(PPStateRunning)
 }
 
-// Errors returns a channel of errors during consumption
-func (pp *PartitionProcessor) Errors() <-chan error {
-	return pp.runnerErrors
-}
-
 // Start initializes the partition processor
 // * recover the table
 // * recover all join tables
 // * run the join-tables in catchup mode
 // * start the processor processing loop to receive messages
-func (pp *PartitionProcessor) Start(ctx context.Context) error {
+func (pp *PartitionProcessor) Start(setupCtx, ctx context.Context) error {
```

Comment:
maybe a comment would be nice to explain the difference between the `setupCtx` and `ctx`, because it's quiet unusual to have two ctxs.

Reply:
totally agree, and it still messes with my head. Added a comment.